src/ds5/advanced_mode/json_loader.hpp

```diff
@@ -442,42 +443,56 @@ namespace librealsense
         return map;
     }
 
-    inline std::vector<uint8_t> generate_json(const preset& in_preset)
+    inline std::vector<uint8_t> generate_json(const device_interface& dev, const preset& in_preset)
     {
         preset_param_group p = in_preset;
         auto fields = initialize_field_parsers(p);
-
-        json j;
+        serialized_utilities::json_preset_writer preset_writer;
+        
+        preset_writer.set_device_info(dev);
+        
         for (auto&& f : fields)
         {
             if (f.second->is_duplicated) // Skip duplicated fields
                 continue;
 
             auto str = f.second->save();
             if (!str.empty()) // Ignored fields return empty string
-                j[f.first.c_str()] = str;
+                preset_writer.write_param(f.first.c_str(), str);
         }
 
-        auto str = j.dump(4);
+        auto str = preset_writer.get_root().dump(4);
```

Comment:
Shouldn't we have something like a `to_string()` call for the writer?

Reply:
good idea 👍